hail_search/hail_search_query.py

```diff
@@ -998,23 +1029,133 @@ def _format_results(self, ht, annotation_fields):
         ht = ht.annotate(selected_transcript=self._selected_main_transcript_expr(ht))
         return super()._format_results(ht, annotation_fields)
 
-    @staticmethod
-    def _gene_ids_expr(ht):
-        return hl.set(ht.sorted_transcript_consequences.map(lambda t: t.gene_id))
-
     @classmethod
     def _omim_sort(cls, r, omim_gene_set):
         return [
             hl.if_else(omim_gene_set.contains(r.selected_transcript.gene_id), 0, 1),
-            -cls._gene_ids_expr(r).intersection(omim_gene_set).size(),
-        ]
+        ] + super()._omim_sort(r, omim_gene_set)
 
     @classmethod
     def _gene_rank_sort(cls, r, gene_ranks):
-        return [
-            gene_ranks.get(r.selected_transcript.gene_id),
-            hl.min(cls._gene_ids_expr(r).map(gene_ranks.get)),
-        ]
+        return [gene_ranks.get(r.selected_transcript.gene_id)] + super()._gene_rank_sort(r, gene_ranks)
+
+
+class SvHailTableQuery(BaseHailTableQuery):
```

Comment:
could these subclasses be in separate files?

Reply:
They could be. I'll make a follow up PR to move everything, to keep the diff more manageable on this and the gcnv PR